### PR TITLE
msbuild: build fix & update to 16.9

### PR DESCRIPTION
**Pull request by nelsontkq**
**Summary**

Many internal changes between 16.8->16.9. Main reason for update is that it's the one mono 16.12.0.* expects.

Full list of changes [available here](https://github.com/dotnet/msbuild/blob/main/documentation/Changelog.md)

solves #2622 

**Test Plan**

 - build a few projects
 - build godot and confirm working

**Checklist**

- [x] Package was built and tested against unstable

### CODE REVIEW

File: packages/m/mono-msbuild/package.yml [hunk 1]
```diff
@@ -13,20 +15,28 @@ description: |
 networking : yes
 builddeps  :
```
Comment (davidjharder) [inline packages/m/mono-msbuild/package.yml:16]: These should be re-ordered, see Actions warning

**Review by davidjharder — CHANGES_REQUESTED** [review]
Thanks for fixing this. One small issue

**Comment by nelsontkq** [comment]
@davidjharder fixed!

**Review by davidjharder — APPROVED** [review]
Thanks!

**Pull request merged**